src/main/kotlin/org/jitsi/nlj/rtcp/KeyframeRequester.kt

```diff
@@ -31,35 +36,94 @@ import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbFirPacketBuilder
  * on what the client supports
  * 3) Aggregation.  This class will pace outgoing requests such that we don't spam the sender
  */
-class KeyframeRequester : ObserverNode("Keyframe Requester") {
+class KeyframeRequester : TransformerNode("Keyframe Requester") {
+
+    private val WAIT_INTERVAL_MS = 100
+
+    override fun transform(packetInfo: PacketInfo): PacketInfo? {
+        val compoundRtcp = packetInfo.packetAs<CompoundRtcpPacket>()
+
+        var drop = false
+        compoundRtcp.packets.forEach { pkt ->
+            when (pkt) {
+                is RtcpFbPacket -> {
+                    val firCommandSeqNum = generateFirCommandSequenceNumber(pkt.mediaSourceSsrc, System.currentTimeMillis())
+                    if (firCommandSeqNum > -1)
+                        when (pkt) {
+                            is RtcpFbFirPacket -> {
+                                if (!hasFirSupport) {
+                                    // NOTE(george): dropping the packet here should work fine as long as we don't
+                                    // receive 2 RTCP packets: one we want to drop and one we want to forward in the
+                                    // same compound packet.
+                                    drop = true
+
+                                    requestKeyframe(pkt.mediaSenderSsrc)
```

Comment:
`requestKeyframe` will also call `generateFirCommandSequenceNumber`, so in the end two seqs will be generated for the same request.

Reply:
We could create a `requestKeyframeInternal(ssrc, firCommandSequenceNumber)` and call that instead. Or do you imagine something different?